junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DefaultLauncher.java

```diff
@@ -58,12 +58,19 @@
 		Preconditions.condition(testEngines != null && testEngines.iterator().hasNext(),
 			() -> "Cannot create Launcher without at least one TestEngine; "
 					+ "consider adding an engine implementation JAR to the classpath");
-		this.testEngines = validateUniqueIds(testEngines);
+		this.testEngines = validateEngineIds(testEngines);
 	}
 
-	private static Iterable<TestEngine> validateUniqueIds(Iterable<TestEngine> testEngines) {
+	private static Iterable<TestEngine> validateEngineIds(Iterable<TestEngine> testEngines) {
 		Set<String> ids = new HashSet<>();
 		for (TestEngine testEngine : testEngines) {
+			// check usage of reserved id prefix
+			if (!validateReservedIds(testEngine)) {
+				logger.warn(() -> String.format("Thou shalt not use an engine ID starting with 'junit-': '%s'",
```

Comment:
The style here is too biblical.

Let's opt for something more like: _Third-party TestEngine implementations are forbidden to use the reserved `junit-` prefix for their ID._